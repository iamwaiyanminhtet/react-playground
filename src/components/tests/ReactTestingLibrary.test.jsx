import {render, screen} from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { UserEvent} from "./ReactTestingLibrary"
import FirstTest from "./ReactTestingLibrary"
import { describe, it, expect } from "vitest"

describe('test component', () => {
    it('heading', () => {
        render(<FirstTest/>)
        expect(screen.getByRole("heading", {name : "First Test"}).textContent).toMatch(/First Test/i)
    })
});

describe('User Event', () => {
    it('initial render', () => {
        const {container} = render(<UserEvent/>)
        expect(container).toMatchSnapshot()
    });

    it("renders radical rhinos after button click", async () => {
        const user = userEvent.setup();

        render(<UserEvent/>);
        const button = screen.getByRole('button', {name : 'Click Me'});

        await user.click(button);
        expect(screen.getByRole('main').textContent).toMatch(/radical rhinos/i);
    })
});