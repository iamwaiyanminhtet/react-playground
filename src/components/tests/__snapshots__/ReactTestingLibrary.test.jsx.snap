// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User Event > initial render 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <main>
    Magnificent Monkeys
  </main>
</div>
`;
